x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/pipeline_processors_editor.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent, useState, useMemo, useEffect, useCallback } from 'react';
+import { EuiButton } from '@elastic/eui';
+
+import { Processor } from '../../../../common/types';
+
+import { OnFormUpdateArg } from '../../../shared_imports';
+
+import { SettingsFormFlyout, DragAndDropTree, PipelineProcessorEditorItem } from './components';
+import { deserialize } from './data_in';
+import { serialize, SerializeResult } from './data_out';
+import { useProcessorsState } from './processors_reducer';
+import { ProcessorInternal, ProcessorSelector } from './types';
+
+interface FormState {
+  validate: OnFormUpdateArg<any>['validate'];
+}
+
+export interface OnUpdateHandlerArg extends FormState {
+  getData: () => SerializeResult;
+}
+
+export type OnUpdateHandler = (arg: OnUpdateHandlerArg) => void;
+
+export interface Props {
+  value: {
+    processors: Processor[];
+    onFailure?: Processor[];
+  };
+  onUpdate: (arg: OnUpdateHandlerArg) => void;
+}
+
+/**
+ * The editor can be in different modes. This enables us to hold
```

Comment:
If I'm understanding the code correctly, the `mode` is really only pertinent to handling logic in the flyout. Maybe we should clarify that here?